### ISSUE TITLE
Bug: Scheme not working correctly after upgrading neovim 0.9.5 to 0.10.0

### ISSUE THREAD

**Issue by theKnightsOfRohan** [joshdick/onedark.vim#333]
- [x] I have carefully read and followed the [Installation section of the README](https://github.com/joshdick/onedark.vim#installation)
- [x] I have searched through the [closed issues](https://github.com/joshdick/onedark.vim/issues?q=is%3Aissue+is%3Aclosed) to see if there is an existing solution for my issue

### Terminal Emulator Info

Terminal emulator(s) and their version(s) used, if applicable: iTerm2, Build 3.4.23

### Output From `vim --version`

```
NVIM v0.10.0
Build type: Release
LuaJIT 2.1.1713773202
Run "nvim -V1 -v" for more info
```

### Issue Description
In neovim 0.10.0, they seem to have renamed a lot of the treesitter highlight groups from 0.9.5, leading to a lot of the colors formerly applied by this colorscheme to revert to white. This also applies to the lsp highlighting. Here is the link to their [article](https://neovim.io/doc/user/news-0.10.html).

**Comment by kuangliu** [comment]
@theKnightsOfRohan try my fixed version: https://github.com/kuangliu/onedark.vim

**Comment by theKnightsOfRohan** [comment]
@kuangliu Unfortunately, the fixed version doesn't work either on 0.10.0.

**Comment by kuangliu** [comment]
That's weird. This is my lazy config:
```
{
  'kuangliu/onedark.vim',
  config = function()
    vim.cmd([[colorscheme onedark]])
    vim.cmd([[set background=dark]])
  end,
},
```
Mine is fine.
![image](https://github.com/joshdick/onedark.vim/assets/10502826/b1b83a76-0e42-4244-bbbf-2cd94bb40593)

**Comment by theKnightsOfRohan** [comment]
The red colors for me don't match @kuangliu.

## 0.9.5 w/ original:
<img width="1680" alt="Screenshot 2024-05-21 at 11 06 33 AM" src="https://github.com/joshdick/onedark.vim/assets/114779675/58c6fd32-d5f7-4b91-8d85-27e952747851">

## 0.10.0 w/ original:
<img width="1680" alt="Screenshot 2024-05-21 at 11 09 25 AM" src="https://github.com/joshdick/onedark.vim/assets/114779675/ce52eb17-fe5c-4107-8130-3b085f93bd28">

## 0.10.0 w/ yours:
<img width="1680" alt="Screenshot 2024-05-21 at 11 10 32 AM" src="https://github.com/joshdick/onedark.vim/assets/114779675/dead8c48-0971-40fb-895d-3515b8699d04">

**Comment by kuangliu** [comment]
@theKnightsOfRohan 
I think that's because I change the "Identifier" color to `s:foreground` instead of `s:red`. 
You can try change it back. It should be the same as before.
![image](https://github.com/joshdick/onedark.vim/assets/10502826/103f2be5-639d-40bd-9f6c-9cb4e3c0564a)

**Comment by theKnightsOfRohan** [comment]
I got it working now with your fork. Thanks!

**Comment by mikesellitto** [comment]
@theKnightsOfRohan I don't think this should be marked at completed, since it's still an issue in the repo (even if the fork fixes it).

@kuangliu would you mind opening a PR with the fixes you made? Thanks for those, by the way!

**Comment by kuangliu** [comment]
Personally, I don't recommend nvim-0.10 users continue using this plugin.
I recently switch to [onedark.nvim](https://github.com/navarasu/onedark.nvim)
```
{
    'navarasu/onedark.nvim',
    config = function()
      local c = require('onedark.palette').dark
      require('onedark').setup({
        highlights = {
          ['@constructor'] = { fg = c.cyan },
          ['@variable.builtin'] = { fg = c.yellow },
          ['@variable.parameter'] = { fg = c.fg },
          ['@lsp.type.parameter'] = { fg = c.fg },
          ['@punctuation.bracket'] = { fg = c.fg },
          ['@punctuation.special'] = { fg = c.blue },
        },
      })
      require('onedark').load()
      vim.api.nvim_set_hl(0, 'NvimTreeNormal', {})
      vim.api.nvim_set_hl(0, 'NvimTreeEndOfBuffer', {})
      vim.api.nvim_set_hl(0, 'NvimTreeRootFolder', { fg = c.green })
    end,
  },
```
I tweaked a little, and it looks pretty much the same.
![image](https://github.com/joshdick/onedark.vim/assets/10502826/31fa07b1-f98c-4761-a9cd-fc67b9ec3d47)

**Comment by theKnightsOfRohan** [comment]
I instead switched to the direct lua port of this plugin and this modified the scheme as I saw fit as well, since I like my transparent background and navarasu's is a little weird about background. The lua port, [ii14/onedark.nvim](https://github.com/ii14/onedark.nvim), is a lot easier of a transition.
